File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/resolve/ImplicitParametersResolveTest.scala
package org.jetbrains.plugins.scala.lang.resolve

import org.jetbrains.plugins.scala.lang.psi.api.base.ScReference
import org.jetbrains.plugins.scala.lang.psi.api.statements.ScFunctionDefinition
import org.junit.Assert._

class ImplicitParametersResolveTest extends ScalaResolveTestCase {

  override def folderPath = super.folderPath + "resolve/implicitParameter"

  def testlocalValAsImplicitParam(): Unit = {
    val ref = findReferenceAtCaret()
    assertTrue(ref.resolve.isInstanceOf[ScFunctionDefinition])
  }

  def testSCL16246(): Unit = {
    findReferenceAtCaret() match {
      case ref: ScReference =>
        assertTrue(ref.multiResolveScala(false).length == 1)
    }
  }

}

File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/resolve/ImplicitParametersResolveTest.java
package org.jetbrains.plugins.scala.lang.resolve;

import com.intellij.psi.PsiReference;
import org.jetbrains.plugins.scala.lang.psi.api.statements.ScFunctionDefinition;

import java.nio.file.Path;

public class ImplicitParametersResolveTest extends ScalaResolveTestCase {
  @Override
  public Path folderPath() {
    return super.folderPath().resolve("resolve").resolve("implicitParameter");
  }

  public void testlocalValAsImplicitParam() {
    PsiReference ref = findReferenceAtCaret();
    assertTrue(ref.resolve() instanceof ScFunctionDefinition);
  }
}
